build/ci/release.yml

```diff
@@ -443,7 +443,7 @@ buildvariants:
   - name: release_mongocli_github
     display_name: "Release MongoCLI (goreleaser)"
     run_on:
-      - rhel80-small
+      - rhel90-small
```

Comment:
[q] why `rhel90` over `rhel94`?

Reply:
we are using rhel90 also for the packaging because I think there are more rhel90 machine in evergreen than rhel94